### PR TITLE
Implement loading screen for cmd-f

### LINKED ISSUE #57: Implement loading screen for cmd-f

### PR DESCRIPTION
**Pull request by yehee**
Fixes #57 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Works on staging site

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by rmehri01 — COMMENTED** [review]
The page was still loading for me when the loading screen stopped and was a bit sudden, maybe there could be a fade like on previous sites? 

Also, for some reason the cmd-f logo looks a bit off:
image

**Comment by yehee** [comment]
@rmehri01 I went to `nwhacks2021` but seems like it's also done through setting timeout. I looked through a couple of examples online, but couldn't seem to get any further :( Do you happen to know any workaround for this?

**Comment by rmehri01** [comment]
@yehee I'm not actually sure about this because I haven't used react much but I think that since you are using the ternary in `index.js`, it only evaluates the false branch at first and then after the timeout is done, then loads the true branch, which would explain why the site still needs to load after the loading screen is done. 

I changed to your branch and tried instead just having `{!loaded && <Loading />}` to basically display the loading screen on top of the actual site and it seemed to work for me. Any thoughts?

**Comment by yehee** [comment]
@rmehri01 Thanks for the suggestion! Though if we do, users can scroll down to see the rest of the site even when the loading screen is displayed on top of it, wouldn't they?

**Comment by rmehri01** [comment]
@yehee Yeah it would need some css changes, kind of like the mobile `Sidebar` so that it is fixed and just fills the whole page.

**Comment by rmehri01** [comment]
Hmm that seems interesting but it is experimental so it says not to use it in production so I'm not really sure.

File: src/pages/index.jsx [hunk 1]
```diff
@@ -21,8 +22,11 @@ export default function Index({
   values,
   configs: { navbarConfig, faqConfig },
 }) {
+  const [loaded, setLoaded] = useState(false)
+  useEffect(() => setTimeout(() => setLoaded(true), 3500), [])
```
Comment (andykwan1124) [inline src/pages/index.jsx:26]: Is 3.5 seconds a bit long? I think around 0.5 seconds would be enough. What do you think?

**Review by andykwan1124 — COMMENTED** [review]
The loading screen looks nice! I see a brief flash of unstyled content though. Wondering if there is a way to fix that?

File: src/pages/index.jsx (hunk 1 above)
Comment (yehee) [reply src/pages/index.jsx:26]: Yeah about that, seems like what @rmehri01 mentioned above, it doesn't start loading the main components until `loaded` evaluates to `true`. Still investigating that but I couldn't find any example online that could be useful in this case :( Seems like you were able to optimize loading time so we can hold onto this for now! Thanks a lot 🙏

**Comment by yehee** [comment]
Closing it for now

**Pull request closed without merging**